### ISSUE TITLE
Controllers Cause Crash on Minecraft versions newer than 1.16.210.05

### ISSUE THREAD

**Issue by theofficialgman** [minecraft-linux/mcpelauncher-manifest#510]
Issue: Minecraft crashes on launcher when controllers are connected to the system (only affects versions of minecraft newer than 1.16.210.05). Does not happen on android.

When controllers are connected to the linux appimage/flatpack distribution launchers, minecraft crashes an launch with the errors similar to below. The same hardware tested (nintendo switch) running android 10 is able to launch minecraft with controllers connected without issue on these minecraft versions which points to this being either a launcher issue or some sort of launcher/game version incompatibility. Controllers can be connected after launch and they function properly, so it it just when they are connected when pressing the PLAY button in the launcher where there is an issue. We have confirmed that is issue is also apparent on the x86 launcher as well.

```
23:53:07 Trace [Launcher] Loading gamepad mappings: /tmp/.mount_MC_3a8vbVCuE/usr/share/mcpelauncher/gamecontrollerdb.txt
23:53:07 Info  [Launcher] Creating window
libevdev_new_from_fd error -9 (/dev/input/event6)
libevdev_new_from_fd error -22 (/dev/input/js1)
libevdev_new_from_fd error -9 (/dev/input/event5)
libevdev_new_from_fd error -22 (/dev/input/js0)
libevdev_new_from_fd error -22 (/dev/input/js2)
23:53:07 Trace [WindowCallbacks] Gamepad connected #0
Signal 11 received
Backtrace elements: 3
#0 /tmp/.mount_MC_3a8vbVCuE/usr/bin/./mcpelauncher-client(_ZN12CrashHandler12handleSignalEiPv+0x19c) [0xcfa844]
#1 linux-vdso.so.1(__kernel_rt_sigreturn+0) [0x7f968546c0]
#2 HYBRIS (null)+0x8 in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e3b1b0 [0x0x7f8bffb1b0]
Dumping stack...
#54 HYBRIS (null)+(nil) in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e3b1a8 [0x0x7f8bffb1a8]
#77 HYBRIS (null)+0x8 in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e3b1b0 [0x0x7f8bffb1b0]
#664 HYBRIS (null)+(nil) in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e3b1a8 [0x0x7f8bffb1a8]
#975 HYBRIS (null)+0x7f8c03d504 in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e7d504 [0x0x7f8c03d504]
#988 HYBRIS (null)+0x7f8c03d47c in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e7d47c [0x0x7f8c03d47c]
#994 HYBRIS (null)+0x7f8c03d47c in /home/garrett/.local/share/mcpelauncher/versions/1.16.221.01/lib/arm64-v8a/libminecraftpe.so+0x0x1e7d47c [0x0x7f8c03d47c]

Process exited with unexpected exit code: 6
```

**Comment by cobalt2727** [comment]
Getting the same thing on a laptop running Ubuntu 20.10 - full logs below to compare:

```
00:33:41 Info  [Launcher] Version: client 992db17 / manifest 8cae128
00:33:41 Info  [Launcher] CPU: GenuineIntel Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
00:33:41 Info  [Launcher] CPU supports SSSE3: YES
00:33:41 Trace [Launcher] Loading hybris libraries
00:33:41 Trace [REDIRECT] /data/data/com.mojang.minecraftpe to /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/
00:33:41 Trace [REDIRECT] /data/data/app/proc/21/cmdline to /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/
00:33:41 Trace [REDIRECT] /data/data/app/bin/mcpelauncher-client to /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/
00:33:41 Trace [REDIRECT] /data/data to /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/
00:33:41 Trace [HybrisUtils] Loaded OS library libm.so.6
00:33:41 Trace [HybrisUtils] Loaded OS library libz.so.1
00:33:41 Trace [HybrisUtils] Loaded OS library /app/share/mcpelauncher/lib/native/x86_64/libfmod.so.12.0
load_library: Undefined symbol glMultMatrixf
load_library: Undefined symbol glFogfv
load_library: Undefined symbol glNormalPointer
load_library: Undefined symbol glVertexPointer
load_library: Undefined symbol glLightfv
load_library: Undefined symbol glLoadIdentity
load_library: Undefined symbol glRotatef
load_library: Undefined symbol glMatrixMode
load_library: Undefined symbol glPopMatrix
load_library: Undefined symbol glTranslatef
load_library: Undefined symbol glOrthof
load_library: Undefined symbol glPushMatrix
load_library: Undefined symbol glEnableClientState
load_library: Undefined symbol glShadeModel
load_library: Undefined symbol glViewport
load_library: Undefined symbol glVertexAttrib2fv
load_library: Undefined symbol glVertexAttrib1fv
load_library: Undefined symbol glValidateProgram
load_library: Undefined symbol glUniformMatrix4fv
load_library: Undefined symbol glUniform4iv
load_library: Undefined symbol glVertexAttrib1f
load_library: Undefined symbol glUniform4fv
load_library: Undefined symbol glUniform4f
load_library: Undefined symbol glUniform3i
load_library: Undefined symbol glAlphaFunc
load_library: Undefined symbol glUniform3fv
load_library: Undefined symbol glUniform2i
load_library: Undefined symbol glScalef
load_library: Undefined symbol glUniform4i
load_library: Undefined symbol glUniform2fv
load_library: Undefined symbol glUniform2f
load_library: Undefined symbol glVertexAttribPointer
load_library: Undefined symbol glUniform1iv
load_library: Undefined symbol glUniform1i
load_library: Undefined symbol glUniform1fv
load_library: Undefined symbol glUniform1f
load_library: Undefined symbol glTexParameteriv
load_library: Undefined symbol glStencilOpSeparate
load_library: Undefined symbol glStencilMaskSeparate
load_library: Undefined symbol glStencilMask
load_library: Undefined symbol glStencilFuncSeparate
load_library: Undefined symbol glFramebufferRenderbuffer
load_library: Undefined symbol glEnable
load_library: Undefined symbol glTexParameteri
load_library: Undefined symbol glBufferSubData
load_library: Undefined symbol glDrawArrays
load_library: Undefined symbol glGetRenderbufferParameteriv
load_library: Undefined symbol glDetachShader
load_library: Undefined symbol glDeleteFramebuffers
load_library: Undefined symbol glDrawElements
load_library: Undefined symbol glDeleteProgram
load_library: Undefined symbol glDepthRangef
load_library: Undefined symbol glCullFace
load_library: Undefined symbol glCreateProgram
load_library: Undefined symbol glDisableVertexAttribArray
load_library: Undefined symbol glColorPointer
load_library: Undefined symbol glDeleteRenderbuffers
load_library: Undefined symbol glPixelStorei
load_library: Undefined symbol glDeleteShader
load_library: Undefined symbol glGetProgramInfoLog
load_library: Undefined symbol glColor4f
load_library: Undefined symbol glUseProgram
load_library: Undefined symbol glClearStencil
load_library: Undefined symbol glShaderBinary
load_library: Undefined symbol glIsShader
load_library: Undefined symbol glReleaseShaderCompiler
load_library: Undefined symbol glBlendEquation
load_library: Undefined symbol glShaderSource
load_library: Undefined symbol glGetVertexAttribPointerv
load_library: Undefined symbol glDepthMask
load_library: Undefined symbol glFogf
load_library: Undefined symbol glVertexAttrib3f
load_library: Undefined symbol glUniformMatrix3fv
load_library: Undefined symbol glGetShaderInfoLog
load_library: Undefined symbol glActiveTexture
load_library: Undefined symbol glTexParameterfv
load_library: Undefined symbol glCopyTexImage2D
load_library: Undefined symbol glStencilOp
load_library: Undefined symbol glClear
load_library: Undefined symbol glIsBuffer
load_library: Undefined symbol glGetBufferParameteriv
load_library: Undefined symbol glDisable
load_library: Undefined symbol glBlendFunc
load_library: Undefined symbol glBlendEquationSeparate
load_library: Undefined symbol glFinish
load_library: Undefined symbol glUniformMatrix2fv
load_library: Undefined symbol glIsTexture
load_library: Undefined symbol glTexSubImage2D
load_library: Undefined symbol glBindFramebuffer
load_library: Undefined symbol glBlendColor
load_library: Undefined symbol glGetUniformLocation
load_library: Undefined symbol glBindBuffer
load_library: Undefined symbol glBlendFuncSeparate
load_library: Undefined symbol glBindTexture
load_library: Undefined symbol glVertexAttrib2f
load_library: Undefined symbol glBindRenderbuffer
load_library: Undefined symbol glUniform3iv
load_library: Undefined symbol glGetVertexAttribiv
load_library: Undefined symbol glDeleteTextures
load_library: Undefined symbol glFlush
load_library: Undefined symbol glClearColor
load_library: Undefined symbol glDeleteBuffers
load_library: Undefined symbol glCreateShader
load_library: Undefined symbol glDepthFunc
load_library: Undefined symbol glGetString
load_library: Undefined symbol glAttachShader
load_library: Undefined symbol glCopyTexSubImage2D
load_library: Undefined symbol glClearDepthf
load_library: Undefined symbol glVertexAttrib3fv
load_library: Undefined symbol glTexParameterf
load_library: Undefined symbol glGetAttribLocation
load_library: Undefined symbol glTexImage2D
load_library: Undefined symbol glLinkProgram
load_library: Undefined symbol glCheckFramebufferStatus
load_library: Undefined symbol glUniform2iv
load_library: Undefined symbol glColorMask
load_library: Undefined symbol glCompileShader
load_library: Undefined symbol glSampleCoverage
load_library: Undefined symbol glEnableVertexAttribArray
load_library: Undefined symbol glCompressedTexSubImage2D
load_library: Undefined symbol glUniform3f
load_library: Undefined symbol glGetTexParameteriv
load_library: Undefined symbol glFramebufferTexture2D
load_library: Undefined symbol glFrontFace
load_library: Undefined symbol glVertexAttrib4fv
load_library: Undefined symbol glGenBuffers
load_library: Undefined symbol glCompressedTexImage2D
load_library: Undefined symbol glGetFloatv
load_library: Undefined symbol glGenFramebuffers
load_library: Undefined symbol glBindAttribLocation
load_library: Undefined symbol glGenRenderbuffers
load_library: Undefined symbol glGenTextures
load_library: Undefined symbol glBufferData
load_library: Undefined symbol glHint
load_library: Undefined symbol glFogx
load_library: Undefined symbol glDisableClientState
load_library: Undefined symbol glGetActiveAttrib
load_library: Undefined symbol glIsRenderbuffer
load_library: Undefined symbol glGenerateMipmap
load_library: Undefined symbol glGetAttachedShaders
load_library: Undefined symbol glGetBooleanv
load_library: Undefined symbol glGetError
load_library: Undefined symbol glVertexAttrib4f
load_library: Undefined symbol glGetProgramiv
load_library: Undefined symbol glGetUniformfv
load_library: Undefined symbol glIsFramebuffer
load_library: Undefined symbol glTexCoordPointer
load_library: Undefined symbol glGetFramebufferAttachmentParameteriv
load_library: Undefined symbol glGetShaderiv
load_library: Undefined symbol glGetIntegerv
load_library: Undefined symbol glGetShaderPrecisionFormat
load_library: Undefined symbol glGetShaderSource
load_library: Undefined symbol glGetUniformiv
load_library: Undefined symbol glGetTexParameterfv
load_library: Undefined symbol glReadPixels
load_library: Undefined symbol glLightModelf
load_library: Undefined symbol glGetVertexAttribfv
load_library: Undefined symbol glIsEnabled
load_library: Undefined symbol glStencilFunc
load_library: Undefined symbol glIsProgram
load_library: Undefined symbol glLineWidth
load_library: Undefined symbol glScissor
load_library: Undefined symbol glGetActiveUniform
load_library: Undefined symbol glPolygonOffset
load_library: Undefined symbol glRenderbufferStorage
00:33:41 Trace [Launcher] Loading Minecraft library
00:33:41 Info  [Launcher] Loaded Minecraft library
00:33:41 Debug [Launcher] Minecraft is at offset 0x0x7f82e0080000
00:33:41 Info  [Launcher] Game version: 0.0.0.0
00:33:41 Info  [Launcher] Applying patches
00:33:41 Debug [CorePatches] Failed to patch, vtable _ZTV21AppPlatform_android23 not found
00:33:41 Info  [Launcher] Initializing JNI
00:33:41 Error [JniSupport] Missing native symbol: Java_com_mojang_minecraftpe_MainActivity_nativeRegisterThis
00:33:41 Error [JniSupport] Missing native symbol: Java_com_mojang_minecraftpe_MainActivity_nativeUnregisterThis
00:33:41 Error [JniSupport] Missing native symbol: Java_com_microsoft_xal_browser_WebView_urlOperationSucceeded
00:33:41 Info  [Launcher] Executing main thread
00:33:41 Trace [Minecraft] Entering JNI_OnLoad 0x7f82e007f700
00:33:41 Trace [Minecraft] JNI_OnLoad completed
00:33:41 Trace [JniSupport] Invoking nativeRegisterThis
00:33:41 Trace [JniSupport] Invoking ANativeActivity_onCreate
00:33:41 Warn  [Main] Android stub called
00:33:41 Warn  [Main] Android stub called
00:33:41 Warn  [Main] Android stub called
00:33:41 Warn  [Main] Android stub called
00:33:41 Info  [Launcher] Loading gamepad mappings
00:33:41 Trace [Launcher] Loading gamepad mappings: /app/share/mcpelauncher/gamecontrollerdb.txt
00:33:41 Trace [Launcher] Loading gamepad mappings: /app/share/mcpelauncher/gamecontrollerdb.txt
00:33:41 Info  [Launcher] Creating window
00:33:41 Trace [WindowCallbacks] Gamepad connected #0
Signal 11 received
Backtrace elements: 3
#0 /app/bin/mcpelauncher-client(_ZN12CrashHandler12handleSignalEiPv+0x12a) [0x6195da]
#1 /usr/lib/x86_64-linux-gnu/libpthread.so.0(+0x13ae0) [0x7f82eb306ae0]
#2 HYBRIS (null)+0x7 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e7 [0x0x7f82e1f8f7e7]
Dumping stack...
#3 HYBRIS (null)+0x7f82e0080000 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x(nil) [0x0x7f82e0080000]
#9 HYBRIS (null)+0x7 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e7 [0x0x7f82e1f8f7e7]
#57 HYBRIS (null)+(nil) in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e0 [0x0x7f82e1f8f7e0]
#74 HYBRIS (null)+(nil) in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e0 [0x0x7f82e1f8f7e0]
#80 HYBRIS (null)+0x7 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e7 [0x0x7f82e1f8f7e7]
#364 HYBRIS (null)+0x7f82e1fd51a2 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f551a2 [0x0x7f82e1fd51a2]
#370 HYBRIS (null)+0x7f82e1fd5100 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55100 [0x0x7f82e1fd5100]
#715 HYBRIS (null)+0x7f82e1fd49b0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f549b0 [0x0x7f82e1fd49b0]
#716 HYBRIS (null)+0x7f82e1fd4a60 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54a60 [0x0x7f82e1fd4a60]
#717 HYBRIS (null)+0x7f82e1fd4b10 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54b10 [0x0x7f82e1fd4b10]
#718 HYBRIS (null)+0x7f82e1fd4bf0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54bf0 [0x0x7f82e1fd4bf0]
#719 HYBRIS (null)+0x7f82e1fd4ca0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54ca0 [0x0x7f82e1fd4ca0]
#720 HYBRIS (null)+0x7f82e1fd48d0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f548d0 [0x0x7f82e1fd48d0]
#721 HYBRIS (null)+0x7f82e1fd4df0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54df0 [0x0x7f82e1fd4df0]
#722 HYBRIS (null)+0x7f82e1fd4e50 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54e50 [0x0x7f82e1fd4e50]
#725 HYBRIS (null)+0x7f82e1fd4f70 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54f70 [0x0x7f82e1fd4f70]
#726 HYBRIS (null)+0x7f82e1fd5030 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55030 [0x0x7f82e1fd5030]
#727 HYBRIS (null)+0x7f82e1fd5040 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55040 [0x0x7f82e1fd5040]
#729 HYBRIS (null)+0x7f82e1fd4d50 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54d50 [0x0x7f82e1fd4d50]
#730 HYBRIS (null)+0x7f82e1fd4da0 in /home/cobalt/.var/app/io.mrarm.mcpelauncher/data/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54da0 [0x0x7f82e1fd4da0]

Process exited with unexpected exit code: 6

```

**Comment by MtagDegen** [comment]
Same thing running a x86_64 laptop with Gentoo.

13:22:36 Info  [Launcher] Version: client 992db17 / manifest 8cae128
13:22:36 Info  [Launcher] CPU: AuthenticAMD AMD Ryzen 7 4700U with Radeon Graphics         
13:22:36 Info  [Launcher] CPU supports SSSE3: YES
13:22:36 Trace [Launcher] Loading hybris libraries
13:22:36 Trace [REDIRECT] /data/data/com.mojang.minecraftpe to /home/marx/.local/share/mcpelauncher/
13:22:36 Trace [REDIRECT] /data/data/tmp/.mount_Minecrauya7R/usr/proc/761/cmdline to /home/marx/.local/share/mcpelauncher/
13:22:36 Trace [REDIRECT] /data/data/tmp/.mount_Minecrauya7R/usr/bin/./mcpelauncher-client to /home/marx/.local/share/mcpelauncher/
13:22:36 Trace [REDIRECT] /data/data to /home/marx/.local/share/mcpelauncher/
13:22:36 Trace [HybrisUtils] Loaded OS library libm.so.6
13:22:36 Trace [HybrisUtils] Loaded OS library libz.so.1
13:22:36 Trace [HybrisUtils] Loaded OS library /tmp/.mount_Minecrauya7R/usr/share/mcpelauncher/lib/native/x86_64/libfmod.so.12.0
13:22:36 Trace [Launcher] Loading Minecraft library
13:22:37 Info  [Launcher] Loaded Minecraft library
13:22:37 Debug [Launcher] Minecraft is at offset 0x0x7fc8faf00000
13:22:37 Info  [Launcher] Game version: 0.0.0.0
13:22:37 Info  [Launcher] Applying patches
13:22:37 Debug [CorePatches] Failed to patch, vtable _ZTV21AppPlatform_android23 not found
13:22:37 Info  [Launcher] Initializing JNI
13:22:37 Error [JniSupport] Missing native symbol: Java_com_mojang_minecraftpe_MainActivity_nativeRegisterThis
13:22:37 Error [JniSupport] Missing native symbol: Java_com_mojang_minecraftpe_MainActivity_nativeUnregisterThis
13:22:37 Error [JniSupport] Missing native symbol: Java_com_microsoft_xal_browser_WebView_urlOperationSucceeded
13:22:37 Info  [Launcher] Executing main thread
13:22:37 Trace [Minecraft] Entering JNI_OnLoad 0x7fc90bfff640
13:22:37 Trace [Minecraft] JNI_OnLoad completed
13:22:37 Trace [JniSupport] Invoking nativeRegisterThis
13:22:37 Trace [JniSupport] Invoking ANativeActivity_onCreate
13:22:37 Warn  [Main] Android stub called
13:22:37 Warn  [Main] Android stub called
13:22:37 Warn  [Main] Android stub called
13:22:37 Warn  [Main] Android stub called
13:22:37 Info  [Launcher] Loading gamepad mappings
13:22:37 Trace [Launcher] Loading gamepad mappings: /tmp/.mount_Minecrauya7R/usr/share/mcpelauncher/gamecontrollerdb.txt
13:22:37 Info  [Launcher] Creating window
libevdev_new_from_fd error -22 (/dev/input/js0)
13:22:37 Trace [WindowCallbacks] Gamepad connected #0
Signal 11 received
Backtrace elements: 3
#0 /tmp/.mount_Minecrauya7R/usr/bin/./mcpelauncher-client(_ZN12CrashHandler12handleSignalEiPv+0xfc) [0xb950ac]
#1 /lib64/libpthread.so.0(+0x126b0) [0x7fc959f3d6b0]
#2 HYBRIS (null)+0x7 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e7 [0x0x7fc8fce0f7e7]
Dumping stack...
#23 HYBRIS (null)+(nil) in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e0 [0x0x7fc8fce0f7e0]
#25 HYBRIS (null)+0x7 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e7 [0x0x7fc8fce0f7e7]
#215 HYBRIS (null)+(nil) in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f0f7e0 [0x0x7fc8fce0f7e0]
#453 HYBRIS renoir::tessellator::TessellateAnimated(renoir::PathBuffer const*, renoir::TessellationParams const&, csl::dyn_array_vector<renoir::SlimVertex, renoir::RenoirStdAllocator<renoir::SlimVertex> >&, csl::dyn_array_vector<unsigned int, renoir::RenoirStdAllocator<unsigned int> >&, renoir::float2&, renoir::float2&, unsigned int&, unsigned int&, renoir::TessellationAlgorithm&)+0x22d0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x5100000 [0x0x7fc900000000]
#473 HYBRIS (null)+0x7fc8fce551a2 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f551a2 [0x0x7fc8fce551a2]
#482 HYBRIS (null)+0x7fc8fce55100 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55100 [0x0x7fc8fce55100]
#488 HYBRIS (null)+0x7fc8fce55100 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55100 [0x0x7fc8fce55100]
#495 HYBRIS (null)+0x7fc8faf00000 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x(nil) [0x0x7fc8faf00000]
#496 HYBRIS (null)+0x7fc8faf00000 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x(nil) [0x0x7fc8faf00000]
#838 HYBRIS (null)+0x7fc8fce549b0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f549b0 [0x0x7fc8fce549b0]
#839 HYBRIS (null)+0x7fc8fce54a60 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54a60 [0x0x7fc8fce54a60]
#840 HYBRIS (null)+0x7fc8fce54b10 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54b10 [0x0x7fc8fce54b10]
#841 HYBRIS (null)+0x7fc8fce54bf0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54bf0 [0x0x7fc8fce54bf0]
#842 HYBRIS (null)+0x7fc8fce54ca0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54ca0 [0x0x7fc8fce54ca0]
#843 HYBRIS (null)+0x7fc8fce548d0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f548d0 [0x0x7fc8fce548d0]
#844 HYBRIS (null)+0x7fc8fce54df0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54df0 [0x0x7fc8fce54df0]
#845 HYBRIS (null)+0x7fc8fce54e50 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54e50 [0x0x7fc8fce54e50]
#848 HYBRIS (null)+0x7fc8fce54f70 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54f70 [0x0x7fc8fce54f70]
#849 HYBRIS (null)+0x7fc8fce55030 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55030 [0x0x7fc8fce55030]
#850 HYBRIS (null)+0x7fc8fce55040 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f55040 [0x0x7fc8fce55040]
#852 HYBRIS (null)+0x7fc8fce54d50 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54d50 [0x0x7fc8fce54d50]
#853 HYBRIS (null)+0x7fc8fce54da0 in /home/marx/.local/share/mcpelauncher/versions/1.16.221.01/lib/x86_64/libminecraftpe.so+0x0x1f54da0 [0x0x7fc8fce54da0]

Process exited with unexpected exit code: 6

**Comment by ChristopherHX** [comment]
Commenting out this line: https://github.com/minecraft-linux/mcpelauncher-client/blob/992db171afcb8f21e34de8b3d3b7948fc66144ab/src/window_callbacks.cpp#L131

Seems to avoid a crash.

**Comment by theofficialgman** [comment]
> Commenting out this line: https://github.com/minecraft-linux/mcpelauncher-client/blob/992db171afcb8f21e34de8b3d3b7948fc66144ab/src/window_callbacks.cpp#L131
> 
> Seems to avoid a crash.

thanks for

**Comment by theofficialgman** [comment]
> Commenting out this line: https://github.com/minecraft-linux/mcpelauncher-client/blob/992db171afcb8f21e34de8b3d3b7948fc66144ab/src/window_callbacks.cpp#L131
> 
> Seems to avoid a crash.

would you consider implementing this before the next release/preview build of the launcher?

**Comment by ChristopherHX** [comment]
v0.2.1 and older has this bug,
v0.2.2 (unreleased will fix it, see [Milestone v0.2.2](https://github.com/minecraft-linux/mcpelauncher-manifest/milestone/2))

_I'm reducing the update cyclus. The launcher is in maintainance mode, the original author (mrarm) isn't using it anymore. There is currently no breaking change which requires a new launcher version.
If you reported this bug, before v0.2.1 it could be fixed already. Building this launcher for macOS is too unstable._

I assume connecting your controller after loading the game works already.

**Comment by theofficialgman** [comment]
yes connecting the controller after the game loads works already.

**Comment by yonderbread** [comment]
> yes connecting the controller after the game loads works already.

Not for me, I can run Minecraft Bedrock Edition version 1.16.210.05 (x86_64) with my controller unplugged, but once the game has started and I'm in a world, I'll plug my DS4 controller in and it crashes.

What's weird is ONLY after it crahes I plug the controller in and try to run the game again, it works...

**Comment by ChristopherHX** [comment]
Released in v0.2.3 yesterday, reopen if it won't work.

> Not for me, I can run Minecraft Bedrock Edition version 1.16.210.05 (x86_64) with my controller unplugged, but once the game has started and I'm in a world, I'll plug my DS4 controller in and it crashes.

Controllers are known to make this launcher more unstable, without providing more information in a new issue deal with it. I don't own a ds4 controller.
